File: src/main/java/tfg/proyecto/TFG/dtos/DTOusuarioModificarSubida.java
package tfg.proyecto.TFG.dtos;

import java.math.BigDecimal;
import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import tfg.proyecto.TFG.modelo.Rol;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DTOusuarioModificarSubida {
	private Long id;
	private String nombre; 
	private String email;
	private String password;
	private LocalDate fechaNacimiento;
	private Rol rol;
	private DTOcuentaBancariaSubida cuenta;
	private Boolean activo;
}

File: appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/data/remote/dto/DTOusuarioModificarSubida.kt
package com.example.appmovilshowpass.data.remote.dto

import com.example.appmovilshowpass.model.Rol
import com.example.appmovilshowpass.model.Usuario
import com.google.gson.annotations.SerializedName
import java.time.LocalDate

data class DTOusuarioModificarSubida (
    val id: Long = 0,
    val nombre: String = "",
    val email: String = "",
    val password: String = "",
    val fechaNacimiento: String = "",
    val foto: String = "",
    val rol: Rol,
    @SerializedName("cuenta") // El campo en el JSON es "cuenta"
    val cuenta: DTOtarjetaBancariaSubida? = null,
    val activo: Boolean = true,
    val reportado: Boolean = false
)

